src/types/sqon.ts

```diff
@@ -1,107 +1,28 @@
 import { z as zod } from 'zod';
+import { matchesSchema } from '../utils/matchesSchema';
+import {
```

Comment:
looks like these types are missing a verbatim import?
```suggestion
import type {
```

Reply:
following up on that, after going further into the file, and I'm honestly not sure how to say this without being mistaken as confrontational... so I'm just going to write my thought process, and hope you know I mean well: 
- I look at line 4 in this file.
- in general terms, I expect `arrayFilter` would be the name for some "function that filters arrays" ("filter" being not very descriptive, but hey... a function that does something, and I know naming things is hard).
- as this is a `types/*` file, naming something `ArrayFilter` reads to me as the type/signature for a function like the one I described above.
- then I look at line 56, where it seems to be used as a template constant for validation in runtime —unless I'm misreading what that function does, which is possible while reviewing in GH; but constants have traditionally been named in upper snake case, e.g. `ARRAY_FILTER.` so maybe it's not a constant either?
- not a function's signature, not a constant, what does this initial capital mean? is that a zod executable schema? why doesn't it just say so in the name instead of having nonstandard casing?
- confoosed, I write this comment. sorry? please don't be upset.
- fin

---

Tl;DR: reading further down into this file, now I'm **not entirely sure of what imports are types, classes/zod objects, or constants**... I find the non-standard capitalization make the code truly hard to read without intellisense.
if it helps illustrate my current understanding of casing standards in JS, this is a somewhat succinct summary: https://medium.com/@sodiq.akanmu001/javascript-casing-conventions-a-must-know-for-every-developer-770a4474a7f0

Reply:
okay, went into the next file (types/sqonFilters), and there it looks like this`arrayFilter` is indeed a zod schema of some sort, and the best I can think of is (and pretty please read this as "what would Justin do, he does not at all mean I have to like it"... not as "why is Justin attacking me, what a jerk, who is he to tell me how to do things" etc. please please please):

- this `arrayFilter` example, being a zod executable object is technically a function.
- that means to me it gets the usual camelCase treatment, and adding `*Schema` makes it clearer what the variable contains. e.g. `arrayFilterSchema`.
- perhaps `arrayFilterZodSchema` or `arrayFilterZSchema` would be better? unclear.
- it's not a type or a class, so I wouldn't use initial capital. that leads to confusion, in scopes/files where Z-usage is not obvious (such as reviewing PRs or onboarding new maintainers).
- it's technically not just a constant either, so upper snake doesn't apply either.

what do you think? am I making any sense?
I'm more than willing to be convinced otherwise if you have good arguments, but I'm just speaking from what I'm used to being convention in JS-land. again, sorry if this upsets you, but what's the point of PR reviews otherwise if not to give honest constructive feedback/thoughts?